src/PowerShellEditorServices.Hosting/Internal/EditorServicesRunner.cs

```diff
@@ -59,14 +68,60 @@ public Task RunUntilShutdown()
             return runAndAwaitShutdown;
         }
 
+        /// <remarks>
+        /// TODO: This class probably should not be <see cref="IDisposable"/> as the primary
+        /// intention of that interface is to provide cleanup of unmanaged resources, which the
+        /// logger certainly is not. Nor is this class used with a <see langword="using"/>. It is
+        /// only because of the use of <see cref="_serverFactory"/> that this class is also
+        /// disposable, and instead that class should be fixed.
+        /// </remarks>
         public void Dispose()
         {
             _serverFactory.Dispose();
         }
 
         /// <summary>
-        /// Master method for instantiating, running and waiting for the LSP and debug servers at the heart of Editor Services.
+        /// This is the servers' entry point, e.g. <c>main</c>, as it instantiates, runs and waits
+        /// for the LSP and debug servers at the heart of Editor Services. Uses <see
+        /// cref="HostStartupInfo"/>.
         /// </summary>
+        /// <remarks>
+        /// The logical stack of the program is:
```

Comment:
@rjmholt Did I get this right?

Reply:
Yeah I think that looks right

Reply:
Thanks! This was super useful (to me) to work through and document.